bundles/plugins/org.bonitasoft.studio.contract/src/org/bonitasoft/studio/contract/ui/wizard/CreateContractInputFromBusinessObjectWizardPage.java

```diff
@@ -36,8 +98,199 @@ protected CreateContractInputFromBusinessObjectWizardPage() {
+        rootNameComposite.setLayout(GridLayoutFactory.fillDefaults().numColumns(2).margins(10, 10).create());
+        rootNameComposite.setLayoutData(GridDataFactory.fillDefaults().grab(true, false).create());
+        final Label label = new Label(rootNameComposite, SWT.NONE);
+        label.setLayoutData(GridDataFactory.fillDefaults().align(SWT.END, SWT.CENTER).create());
+        label.setText(Messages.rootContractInputName);
+        final Text prefixText = new Text(rootNameComposite, SWT.BORDER);
+        prefixText.setLayoutData(GridDataFactory.fillDefaults().grab(true, true).create());
+        final IObservableValue prefixObservable = PojoObservables.observeValue(this, "rootName");
+        dbc.bindValue(prefixObservable,
+                EMFObservables.observeDetailValue(Realm.getDefault(), selectedDataObservable, ProcessPackage.Literals.ELEMENT__NAME),
+                neverUpdateValueStrategy().create(), updateValueStrategy().withConverter(dataNameToRootContractInputName()).create());
+        dbc.bindValue(SWTObservables.observeText(prefixText, SWT.Modify),
+                prefixObservable, updateValueStrategy().withValidator(uniqueValidator().onProperty("name").in(contract.getInputs())).create(),
+                null);
+    }
+
+    /**
+     * @return
+     */
+    private IConverter dataNameToRootContractInputName() {
+        return new Converter(String.class, String.class) {
+
+            @Override
+            public Object convert(final Object fromObject) {
+                final String className = (String) EMFObservables.observeDetailValue(Realm.getDefault(), selectedDataObservable,
+                        ProcessPackage.Literals.JAVA_OBJECT_DATA__CLASS_NAME).getValue();
+                if (className != null) {
+                    final String name = fromObject + NamingUtils.getSimpleName(className);
+                    return NamingUtils.generateNewName(newHashSet(transform(contract.getInputs(), toContactInputName())), name, 0);
+                }
+                return "";
+            }
+        };
+    }
+
+    private Function<ContractInput, String> toContactInputName() {
+        return new Function<ContractInput, String>() {
+
+            @Override
+            public String apply(final ContractInput input) {
+                return input.getName();
+            }
+        };
+    }
+
+    private void createProcessDataMappingTreeViewer(final Composite composite, final EMFDataBindingContext dbc) {
+        treeViewer = new CheckboxTreeViewer(composite, SWT.FULL_SELECTION | SWT.BORDER | SWT.V_SCROLL);
+        treeViewer.getTree().setLayoutData(GridDataFactory.fillDefaults().grab(true, true).hint(SWT.DEFAULT, 200).create());
+        treeViewer.getTree().setHeaderVisible(true);
+        treeViewer.addCheckStateListener(this);
+        treeViewer.setCheckStateProvider(this);
+        treeViewer.setContentProvider(new ObservableListTreeContentProvider(new IObservableFactory() {
```

Comment:
isn't a quite generic ObservableListTreeContentProvider, it migth worth extracting it and/or extracting the classes in arguments